nixos/modules/services/databases/couchdb.nix

```diff
@@ -134,6 +151,13 @@ in {
           Extra configuration. Overrides any other configuration.
```

Comment:
I also don't know how you feel about converting `extraConfig` to be of type `attrs`, and then we can use `lib.generators.toINI {} cfg.extraConfg` to convert it.

It'd be a breaking change which is why I refrained from making it, but it feels more "nixy" to me.

Reply:
It would be a breaking change, yes. You can change it if you want to. (Breaking changes are only limited a few weeks per year.)

Progress towards converting all extraConfig-like attributes is tracked here: #144575

Reply:
Let me know if you want to change it in this PR :)

If not I will merge as is.

Reply:
Neat. I didn't know there was an RFC to do that.

I squashed all my commits related to additional config files into a single commit, and added a new one for RFC42. Would you like me to squash that one as well so that this entire PR is a single commit? Or would you like me to keep them separate since it is it's own unique initiative?

Reply:
It works well as another commit. Please make sure to add `mkRemovedOptionModule` for the old option.

Reply:
I didn't remove any options, I changed the type of the existing option. Should I have created a new option instead?

Reply:
Oh right, I must have misread that. It works fine like this I think.